### PR TITLE
Windows development

### PR DESCRIPTION
**Pull request by shepherd007**
These changes are touching code that is common for all supported systems. Actual Visual C++ project will be added in the next step.

Change log:
- added POSIX function stubs to the 'windows' folder
- added some ifdefs for signals usage
- added some ifdefs for network operations
- some tabs in the existing code was replaced with spaces for consistency

Notes:
- pthreads library is going to be downloaded directly by Visual (via Nuget mechanism) - there is no need to add stubs to the repo

### CODE REVIEW

File: dump1090.c [hunk 1]
```diff
@@ -738,6 +744,40 @@ MODES_DUMP1090_VARIANT " " MODES_DUMP1090_VERSION
...
+        "\n"
+        " For further details refer to <https://github.com/mutability/dump1090>\n"
+        "\n"
+    );
+
+    // delay for 1 second to give the user a chance to read the copyright
+    while (llTime >= time(NULL)) {}
+}
+#endif
+
```
Comment (mutability) [inline dump1090.c]: Why is this needed?

File: windows/rtl-sdr.h [hunk 2]
```diff
@@ -0,0 +1,366 @@
+/*
```
Comment (mutability) [inline windows/rtl-sdr.h]: Don't include a copy of headers available upstream.

**Comment by mutability** [comment]
A couple of general comments:
- Not a fan of the #ifdefs. In an ideal world the #ifdefs would be limited to including a different set of headers in one place only e.g. in dump1090.h. The biggest set seems to be in anet.c; you might want to look at completely separate implementations for POSIX vis Windows; where there are POSIX dependencies in the main code (e.g. use of close()) pull those out into a wrapper function that has alternative implementations rather than an #ifdef within the main code.
- Whitespace changes get in the way of seeing what is actually changed; please avoid changed to whitespace mingled with other changes. If you want to send through a separate PR with just whitespace changes, that's fine.

File: dump1090.c (hunk 1 above)
Comment (shepherd007) [inline dump1090.c]: Added to be more compliant with MalcolmRobb/dump1090 repository. Now I have mixed feelings about that - do you want me to remove displaying copyrights from dump1090 and view1090?

Comment (mutability) [inline dump1090.c]: Well:
- There's no requirement in either the BSD or GPL licenses to display this at runtime;
- It's the wrong license text anyway since this fork is GPL;
- There's nothing Windows-specific about this.

Blindly copying stuff from the MalcolmRobb fork is a bad idea.

Comment (shepherd007) [inline dump1090.c]: Sounds reasonable - I'll remove that.

File: windows/rtl-sdr.h (hunk 2 above)
Comment (shepherd007) [inline windows/rtl-sdr.h]: Agree. To be removed.

**Comment by shepherd007** [comment]
Applied requested changes:
- anet.c has windows implementation in separate file
- whitespace changes were reverted and now are not part of this commit
- removed showCopyright() artifacts from all files

File: dump1090.c [hunk 3]
```diff
@@ -922,9 +928,13 @@ int main(int argc, char **argv) {
     // Set sane defaults
     modesInitConfig();
 
+#ifdef WIN32
```
Comment (shepherd007) [inline dump1090.c]: @mutability I would propose replacing both windows and posix section with function call:
initSignals(&Mode.exit);

File: dump1090.c [hunk 4]
```diff
@@ -82,6 +82,7 @@ static void log_with_timestamp(const char *format, ...)
     fprintf(stderr, "%s  %s\n", timebuf, msg);
 }
 
+#ifndef WIN32
 static void sigintHandler(int dummy) {
```
Comment (shepherd007) [inline dump1090.c]: @mutability I am thinking about moving both windows and posix implementations to the separate files (i.e. signaling.c). Example of implementation would be like:
posix:
static int* exit_flag = NULL;

static void sigintHandler(int dummy) 
{
 ...
}
void initSignals(int* exit)
{
    exit_flag = exit;
     signal(SIGINT, sigintHandler);
     signal(SIGTERM, sigtermHandler);
}

windows:
static int* exit_flag = NULL;

BOOL WINAPI HandlerRoutine(DWORD dwCtrlType)
{
   ...
}

void initSignals(int* exit)
{
    exit_flag = exit;
   SetConsoleCtrlHandler(HandlerRoutine, TRUE);
}

**Comment by shepherd007** [comment]
Cleaned utils.c and cprtest.c from WIN32 defines - it is being handled without preprocesor.

**Comment by shepherd007** [comment]
- decided to split net_io.c specific for windows and posix. Posix version has removed defs for WIN32
- moved logging of messages to separate logging.c file
- moved signal handlers to separate signals.c file
- moved console related functions to separate console.c file
